src/inmanta/loader.py

```diff
@@ -95,10 +95,11 @@ def get_siblings(self) -> Iterator["SourceInfo"]:
     @property
     def requires(self) -> List[str]:
         """List of python requirements associated with this source file"""
-        from inmanta.module import Project
+        from inmanta.module import Module, ModuleV1, Project
 
         if self._requires is None:
-            self._requires = Project.get().modules[self._get_module_name()].get_python_requirements_as_list()
+            module: Module = Project.get().modules[self._get_module_name()]
+            self._requires = module.get_python_requirements_as_list() if isinstance(module, ModuleV1) else []
```

Comment:
would it not be cleaner to have this method on both module types, where a v2 always return []?

Reply:
This is something I went back and forth on. I preferred this because it keeps the `ModuleV2` interface free from legacy methods that will eventually get pruned. But I'm willing to change this if you think otherwise.

Reply:
Though I now see that Arnaud has already added this to `ModuleV2` where it returns its `setup.cfg` requirements. I'll have to investigate the consequences. The way `SourceInfo.requires` is used, we probably don't want this but I'm not sure yet.

Reply:
I like it to be clean now. ModuleV1 will be around for quite a bit into the future and we do have to maintain it.